tests/common/test_signalctl.py

```diff
@@ -0,0 +1,388 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2021-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import asyncio
+import contextlib
+import inspect
+import os
+import pickle
+import signal
+import subprocess
+import sys
+import tempfile
+import textwrap
+
+from edb.common import signalctl
+from edb.testbase import server as tb
+
+
+@contextlib.asynccontextmanager
+async def spawn(method):
```

Comment:
I like magic, but this might be a bit too much. Is there any advantages over the approach we use in uvloop? https://github.com/MagicStack/uvloop/blob/master/tests/test_signals.py#L142

Reply:
The advantage is very limited, like being covered by mypy and flake.